### PR TITLE
Update base image and leeway

### PR DESCRIPTION
**Pull request by aledbf**
## Related Issue(s)
Required for: https://github.com/gitpod-io/gitpod/pull/14524

## Release Notes
```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [x] /werft with-preview
- [x] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by roboquat** [comment]
@aledbf: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Review by mads-hartmann — CHANGES_REQUESTED** [review]
Thanks for upgrading Leeway and the base image!  The Leeway version bump comes with breaking changes. CI won't catch it right now due to job protection ([see issue](https://github.com/gitpod-io/ops/issues/3723)) so you'll have to start a job manually using

```
werft job run github
```

I did that [here](https://werft.gitpod-dev.com/job/gitpod-build-aledbf-leeway.3) and it fails before it even gets to the build as there is apparently a breaking change made to `previewctl` [here](https://github.com/gitpod-io/gitpod/pull/14241) - our image was never rebuilt and updated after that change it seems ☹️  We have an issue to not include `previewctl` in the docker image to avoid hickups like this in the [future](https://github.com/gitpod-io/gitpod/issues/14149).

If you want to try to fix all the breaking changes in this PR then:

1. For the leeway breaking change It should be enough just to remove `--dont-retag` everywhere.
2. For `previewctl` all `--kube-save-path` has to be replaced with invocations where `KUBECONFIG` is set instead.

If you'd rather not deal with this mess I understand and I can take care of it, but I won't have time until Monday.

File: dev/preview/BUILD.yaml [hunk 1]
```diff
@@ -15,7 +15,7 @@ scripts:
   - name: get-credentials
     description: Provisions a new preview environment
     script: |
-      previewctl get-credentials --kube-save-path $HOME/.kube/config
+      previewctl get-credentials KUBECONFIG=$HOME/.kube/config
```
Comment (mads-hartmann) [inline dev/preview/BUILD.yaml:18]: The `KUBECONFIG=$HOME/.kube/config` should be before the command execution, but it looks like this leeway script isn't used anywhere and the description of it is wrong, so we can clean this up some other time ☺️

**Review by mads-hartmann — APPROVED** [review]
Thanks! Works like a charm. Tested in-workspace using `leeway run dev/preview:build` and Werft job using `werft job run github` [here](https://werft.gitpod-dev.com/job/gitpod-build-aledbf-leeway.5).

I left one comment, but it doesn't block merge so approving 🚀

File: dev/preview/BUILD.yaml (hunk 1 above)
Comment (aledbf) [reply dev/preview/BUILD.yaml]: fixed

**Review by jenting — APPROVED** [review]

**Review by jeanp413 — APPROVED** [review]

**Review by ArthurSens — APPROVED** [review]

**Pull request merged**